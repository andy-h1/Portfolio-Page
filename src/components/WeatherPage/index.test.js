import React from 'react';
import axiosMock from 'axios';
import { render, cleanup, waitFor } from '@testing-library/react';
import { WeatherPage } from '.';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

jest.mock('axios');

const mockData = {
  dt: 1598656029,
  main: {
    temp: 27.9,
    feels_like: 25.9
  },
  name: 'London',
  sys: {
    country: 'GB'
  },
  weather: [
    {
      description: 'few clouds',
      icon: '02d',
      main: 'Clouds'
    }
  ],
  wind: {
    speed: 5.0
  }
};

describe('Weather Page Component', () => {
  it('should render', () => {
    const { container } = render(<WeatherPage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should fetch and return weather data', async () => {
    axiosMock.get.mockResolvedValue({ data: mockData });
    const { getByTestId } = render(<WeatherPage />);
    await waitFor(() => expect(getByTestId('weatherTracker')));
  });
});

// mock an API call to OpenWeather API
// fake data based on API - does it need to be in the same format?
// render the data
//  expect(getByTestId('loading')).toHaveTextContent('Loading...')
