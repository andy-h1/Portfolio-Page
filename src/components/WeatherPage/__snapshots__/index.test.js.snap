// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Page Component should render 1`] = `
<div
  class="sc-AxjAm jFllak"
>
  <h2
    class="sc-AxirZ jCgDWR"
  >
    Current Weather
  </h2>
  <h3
    data-testid="loading"
  >
    Loading...
  </h3>
  
  <p>
    Change location:
  </p>
  <form
    class="sc-AxiKw bhiawi"
  >
    <label
      class="sc-AxhCb fmuXyx"
      for="cityInput"
    >
      <input
        class="sc-AxgMl eaiYi"
        name="city"
        placeholder="Enter city"
        type="text"
      />
    </label>
    <label
      class="sc-AxhCb fmuXyx"
      for="countryInput"
    >
      <input
        class="sc-AxgMl eaiYi"
        name="country"
        placeholder="Enter country"
        type="text"
      />
    </label>
    <label
      class="sc-AxhUy enhySx"
      for="unitInput"
    >
      <input
        checked=""
        name="units"
        type="radio"
        value="metric"
      />
      Celsius
    </label>
    <label
      class="sc-AxhUy enhySx"
      for="unitInput"
    >
      <input
        name="units"
        type="radio"
        value="imperial"
      />
      Fahrenheit
    </label>
    <button
      class="sc-AxheI fzLXlG"
      type="button"
    >
      Get Weather
    </button>
  </form>
</div>
`;
